.changeset/clean-queens-judge.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend': patch
+---
+
+Refactored catalog collator, extracted abstract class and entity processor to make it possible to re-use functionality.
```

Comment:
The proposed solution no longer uses an abstract class, so this should be updated.  Could be nice to rephrase so the technical details are removed, and then link off to docs page for more details.  Something like:

> The process of adding or modifying fields in the `software-catalog` search index has been simplified. For more details, see [how to add or modify fields in the software catalog index](https://backstage.io/docs/features/search/how-to-guides#how-to-add-or-modify-fields-in-the-software-catalog-index).

Naturally, a corresponding section [in the docs](https://github.com/backstage/backstage/blob/master/docs/features/search/how-to-guides.md) would be necessary.  Would be good to document...

- How to do simple field additions or modifications
- How to rewrite the transformation entirely
- Highlighting that `location` is modified a different way

Reply:
I think ability to create custom entity document was already here, even it was possible to customize document for specific kind of catalog, this PR just simplifies it, so basically no brand new functionality

Reply:
updated changeset